pkg/webhook/webhook.go

```diff
@@ -172,9 +190,39 @@ func (w *Webhook) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 			return
 		}
 		for _, gitjob := range gitjobs {
+			if gitjob.Spec.Git.Revision != "" {
+				continue
+			}
+
 			if !repoRegexp.MatchString(gitjob.Spec.Git.Repo) {
 				continue
 			}
+
+			// if onTag is enabled, we only watch tag event, as it can be coming from any branch
+			if gitjob.Spec.Git.OnTag != "" {
+				if tag == "" {
+					continue
+				}
```

Comment:
What scenarios would OnTag be non-empty and tag would be empty? Accidental? We might want to log it.

Reply:
Quite often, If tag is empty then that means the current event is not a tag event. And we don't want to process the gitjob. So skipping. There shouldn't be a need to log this.

Reply:
This is a nit then: would it be alright to add a comment explaining that?